skimage/morphology/convex_hull.py

```diff
@@ -34,48 +46,50 @@ def convex_hull_image(image):
-
-    if Delaunay is None:
-        raise ImportError("Could not import scipy.spatial.Delaunay, "
-                          "only available in scipy >= 0.9.")
+    ndim = image.ndim
 
-    # Here we do an optimisation by choosing only pixels that are
+    # In 2D, we do an optimisation by choosing only pixels that are
     # the starting or ending pixel of a row or column.  This vastly
     # limits the number of coordinates to examine for the virtual hull.
-    coords = possible_hull(image.astype(np.uint8))
-    N = len(coords)
+    if ndim == 2:
+        coords = possible_hull(image.astype(np.uint8))
+    else:
+        coords = np.transpose(np.nonzero(image))
+        if offset_coordinates:
+            # when offsetting, we multiply number of vertices by 2 * ndim.
+            # therefore, we reduce the number of coordinates by using a
+            # convex hull on the original set, before offsetting.
+            hull0 = ConvexHull(coords)
+            coords = hull0.points[hull0.vertices]
 
     # Add a vertex for the middle of each pixel edge
-    coords_corners = np.empty((N * 4, 2))
-    for i, (x_offset, y_offset) in enumerate(zip((0, 0, -0.5, 0.5),
-                                                 (-0.5, 0.5, 0, 0))):
-        coords_corners[i * N:(i + 1) * N] = coords + [x_offset, y_offset]
+    if offset_coordinates:
+        offsets = _offsets_diamond(image.ndim)
+        coords = (coords[:, np.newaxis, :] + offsets).reshape(-1, ndim)
 
     # repeated coordinates can *sometimes* cause problems in
-    # scipy.spatial.Delaunay, so we remove them.
-    coords = unique_rows(coords_corners)
-
-    # Subtract offset
-    offset = coords.mean(axis=0)
-    coords -= offset
+    # scipy.spatial.ConvexHull, so we remove them.
+    coords = unique_rows(coords)
 
     # Find the convex hull
-    chull = Delaunay(coords).convex_hull
-    v = coords[np.unique(chull)]
-
-    # Sort vertices clock-wise
-    v_centred = v - v.mean(axis=0)
-    angles = np.arctan2(v_centred[:, 0], v_centred[:, 1])
-    v = v[np.argsort(angles)]
-
-    # Add back offset
-    v += offset
-
-    # For each pixel coordinate, check whether that pixel
-    # lies inside the convex hull
-    mask = grid_points_in_poly(image.shape[:2], v)
+    hull = ConvexHull(coords)
+    vertices = hull.points[hull.vertices]
+
+    # If 2D, sort vertices clock-wise, and use Cython function to locate
+    # convex hull pixels
+    if ndim == 2:
+        offset = np.mean(vertices, axis=0)
+        v_centred = vertices - offset
+        angles = np.arctan2(v_centred[:, 0], v_centred[:, 1])
+        vertices = vertices[np.argsort(angles)]
+        mask = grid_points_in_poly(image.shape, vertices)
+    else:
+        gridcoords = np.reshape(np.mgrid[tuple(map(slice, image.shape))],
+                                (ndim, -1))
+        # A point is in the hull if it satisfies all of the hull's inequalities
+        coords_in_hull = np.all(hull.equations[:, :ndim].dot(gridcoords) +
```

Comment:
By any chance, have you checked how much this is slower compared to `grid_points_in_poly`?

Reply:
You mean in 2D? No, I haven't. You're right, worth a look.

Reply:
@soupault the hull equations are about 4.9x slower than `grid_points_in_poly`. There might be some optimization to be done with the array order of the coordinates, but it escapes me at present.

Reply:
Here is my test with `horse`, but the timing results were remarkably consistent when using `measure.tests.test_regionprops.SAMPLE`.

```python
In [26]: %timeit grid_points_in_poly2(image.shape, hull)
10.7 ms ± 199 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [27]: %timeit ch.grid_points_in_poly(image.shape, vertices)
2.16 ms ± 50.4 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Reply:
@jni Huge thanks for the benchmarking!